src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+        if (errorBlob)
+        {
+            std::cout << "D3D11: With message: " << static_cast<const char*>(errorBlob->GetBufferPointer()) << "\n";
+        }
+        return false;
+    }
+    
+    if (FAILED(_device->CreateVertexShader(
+        vertexShaderBlob->GetBufferPointer(),
+        vertexShaderBlob->GetBufferSize(),
+        nullptr,
+        &_vertexShader)))
+    {
+        std::cout << "D3D11: Failed to compile vertex shader\n";
+        return false;
+    }
+    if (FAILED(_device->CreatePixelShader(
+        pixelShaderBlob->GetBufferPointer(),
+        pixelShaderBlob->GetBufferSize(),
+        nullptr,
+        &_pixelShader)))
+    {
+        std::cout << "D3D11: Failed to compile pixel shader\n";
+        return false;
+    }
+    constexpr D3D11_INPUT_ELEMENT_DESC vertexInputLayoutInfo[] =
+    {
+        { "POSITION", 0, DXGI_FORMAT_R32G32B32_FLOAT, 0, offsetof(VertexFormat, vertex), D3D11_INPUT_PER_VERTEX_DATA, 0},
+        { "COLOR", 0, DXGI_FORMAT_R32G32B32_FLOAT, 0, offsetof(VertexFormat, color), D3D11_INPUT_PER_VERTEX_DATA, 0},
+    };
+    if (FAILED(_device->CreateInputLayout(
+        vertexInputLayoutInfo,
+        _countof(vertexInputLayoutInfo),
+        vertexShaderBlob->GetBufferPointer(),
+        vertexShaderBlob->GetBufferSize(),
+        &_vertexLayout)))
+    {
+        std::cout << "D3D11: Failed to create default vertex input layout\n";
+    }
+    // This section to create the vertex and index buffer of the triangle
+    const VertexFormat vertices[] =
+    {
+        //         Vertex                  Color
+        { {  0.0f,  0.5f, 0.0f }, { 0.25f, 0.39f, 0.19f } },
+        { {  0.5f, -0.5f, 0.0f }, { 0.44f, 0.75f, 0.35f } },
+        { { -0.5f, -0.5f, 0.0f }, { 0.38f, 0.55f, 0.20f } },
+    };
+    D3D11_BUFFER_DESC bufferInfo = {};
+    bufferInfo.ByteWidth = sizeof vertices;
+    bufferInfo.Usage = D3D11_USAGE_IMMUTABLE;
+    bufferInfo.BindFlags = D3D11_BIND_VERTEX_BUFFER;
+    D3D11_SUBRESOURCE_DATA resourceData = {};
+    resourceData.pSysMem = vertices;
+    if (FAILED(_device->CreateBuffer(
+        &bufferInfo,
+        &resourceData,
+        &_triangleVerts)))
+    {
+        std::cout << "D3D11: Failed to create triangle vertex buffer\n";
+        return false;
+    }
+    const uint32_t indices[] =
+    {
+        0, 1, 2
+    };
+    bufferInfo.ByteWidth = sizeof indices;
+    bufferInfo.BindFlags = D3D11_BIND_INDEX_BUFFER;
+    resourceData.pSysMem = indices;
+    if (FAILED(_device->CreateBuffer(
+        &bufferInfo,
+        &resourceData,
+        &_triangleIndices)))
+    {
+        std::cout << "D3D11: Failed to create index buffer\n";
+        return false;
+    }
+    return true;
+}
+
+void HelloCOMApplication::OnResize(GLFWwindow* window, int32_t width, int32_t height)
+{
+    // The potential errors that happen in here should propagate somewhere maybe?
+    auto* application = static_cast<HelloCOMApplication*>(glfwGetWindowUserPointer(window));
```

Comment:
Yeah if we can hide that away in Application that would be neat. 

and introduce Setters for Width/Height in Application might make sense too now that I am looking at this blob here.